src/main/java/com/minecolonies/coremod/client/gui/WindowMultiBlock.java

```diff
@@ -0,0 +1,295 @@
+     */
+    private static final String RES_STRING = "textures/gui/buildtool/%s.png";
+
+    /**
+     * Green String for selected left click.
+     */
+    private static final String GREEN_POS = "_green";
+
+    /**
+     * Red String for selected right click.
+     */
+    private static final String RED_POS = "_red";
+
+    /**
+     * Position of the multiblock.
+     */
+    private final BlockPos pos;
+
+    /**
+     * The direction it is facing.
+     */
+    private EnumFacing facing = UP;
+
+    /**
+     * The output direction.
+     */
+    private EnumFacing output = DOWN;
+
+    /**
+     * The player which opened the GUI.
+     */
+    private final EntityPlayer player = Minecraft.getMinecraft().player;
+
+    /**
+     * The input field with the range.
+     */
+    private final TextField inputRange;
+
+    /**
+     * The input field with the range.
+     */
+    private final TextField inputSpeed;
+
+    /**
+     * The constructor called before opening this window.
+     * @param pos the position of the TileEntity which this window belogs to.
+     */
+    public WindowMultiBlock(@Nullable final BlockPos pos)
+    {
+        super(Constants.MOD_ID + MULTI_BLOCK_RESOURCE_SUFFIX);
+        this.pos = pos;
+        inputRange = findPaneOfTypeByID(INPUT_RANGE_NAME, TextField.class);
+        inputSpeed = findPaneOfTypeByID(INPUT_SPEED, TextField.class);
+        this.init();
+    }
+
+    private void init()
+    {
+        //Register all necessary buttons with the window.
+        registerButton(BUTTON_CONFIRM, this::confirmClicked);
+        registerButton(BUTTON_CANCEL, this::cancelClicked);
+        registerButton(BUTTON_LEFT, this::moveLeftClicked);
+        registerButton(BUTTON_RIGHT, this::moveRightClicked);
+        registerButton(BUTTON_UP, this::moveUpClicked);
+        registerButton(BUTTON_DOWN, this::moveDownClicked);
+        registerButton(BUTTON_BACKWARD, this::moveBackClicked);
+        registerButton(BUTTON_FORWARD, this::moveForwardClicked);
+    }
+
+    /**
+     * Called when the window is opened.
+     * Sets up the buttons for either hut mode or decoration mode.
+     */
+    @Override
+    public void onOpened()
+    {
+        final TileEntity block = Minecraft.getMinecraft().world.getTileEntity(pos);
+        if (block instanceof TileEntityMultiBlock)
+        {
+            inputRange.setText(Integer.toString(((TileEntityMultiBlock) block).getRange()));
+            inputSpeed.setText(Integer.toString(((TileEntityMultiBlock) block).getSpeed()));
+            final EnumFacing dir = ((TileEntityMultiBlock) block).getDirection();
+            final EnumFacing out = ((TileEntityMultiBlock) block).getOutput();
+            enable(dir, dir, false);
+            enable(out, out, true);
+            return;
+        }
+        close();
+    }
+
+    private void enable(final EnumFacing oldFacing, final EnumFacing newFacing, final boolean rightClick)
+    {
+        switch (oldFacing)
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a default case to this switch. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ASwitchLastCaseIsDefaultCheck)